packages/material-ui/src/Stack/Stack.js

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  createUnaryUnit,
+  handleBreakpoints,
+  unstable_extendSxProp as extendSxProp,
+} from '@material-ui/system';
+import { getValue } from '@material-ui/system/spacing';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+
+const getSideFromDirection = (direction) => (direction === 'row' ? 'Left' : 'Top');
+
+const StackRoot = experimentalStyled(
+  'div',
+  {},
+  { name: 'Stack' },
+)(({ styleProps, theme }) => {
+  let styles = {
+    display: 'flex',
+    flexDirection: styleProps.direction,
+  };
+
+  if (styleProps.spacing) {
+    const transformer = createUnaryUnit(theme, 'spacing', 8, 'spacing');
+    const styleFromPropValue = (propValue) => ({
+      '& > * + *': {
```

Comment:
What do you think about:

```suggestion
      '& > :not(styles) + :not(styles)': {
```

to increase the CSS specificity for children that already have a margin CSS property and for hedging against emotion that inject `<style>` element in the body (this one, we aim to not support it, but it seems to come for free).